### PR TITLE
generate an uber-pull-secret

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Begins [DPTP-1845](https://issues.redhat.com/browse/DPTP-1845)

/assign @alvaroaleman @petr-muller

### CODE REVIEW

File: core-services/ci-secret-bootstrap/_config.yaml [hunk 1]
```diff
@@ -2311,6 +2311,249 @@ secret_configs:
         namespace: test-credentials
         name: registry-pull-credentials
         type: kubernetes.io/dockerconfigjson
+  # This secret contains the pull secrets for all possible registries.
+  - from:
+      .dockerconfigjson:
+        dockerconfigJSON:
```
Comment (alvaroaleman) [inline core-services/ci-secret-bootstrap/_config.yaml:2317]: how are we going to keep all of those in sync?

Comment (stevekuznetsov) [reply core-services/ci-secret-bootstrap/_config.yaml]: done

File: hack/generate-pull-secret-entries.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
```
Comment (alvaroaleman) [inline hack/generate-pull-secret-entries.py:1]: Can we make this something that just validates that all .dockerconfigjson for `cluster-pul-secret` entries are identical except for the local cluster? This now rewrites the whole config

Comment (stevekuznetsov) [reply hack/generate-pull-secret-entries.py:1]: What is wrong with the current approach, specifically? It's not *only* that they need to be identical, they *also* must have correct content.

Comment (stevekuznetsov) [reply hack/generate-pull-secret-entries.py:1]: We auto-generate and re-format every other file in this repo, I can't see how another is any worse (with a convenient `make` target to do it, too)

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**